src/EditorFeatures/CSharpTest/Extensions/ContextQuery/IsPossibleDeconstructionDesignationTests.cs

```diff
@@ -0,0 +1,157 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System.Threading;
+using Microsoft.CodeAnalysis.CSharp.Extensions.ContextQuery;
+using Roslyn.Test.Utilities;
+using Xunit;
+using Microsoft.CodeAnalysis.Shared.Extensions;
+
+namespace Microsoft.CodeAnalysis.Editor.CSharp.UnitTests.IntelliSense.CompletionSetSources
+{
+    public class IsPossibleDeconstructionDesignationTests : AbstractContextTests
+    {
+        protected override void CheckResult(bool expected, int position, SyntaxTree syntaxTree)
+        {
+            var actual = syntaxTree.IsPossibleDeconstructionDesignation(position, CancellationToken.None);
+            Assert.Equal(expected, actual);
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.Completion)]
+        public void Test1()
+        {
+            VerifyTrue(AddInsideMethod(@"(var $$, var y)"));
```

Comment:
Do these test with all important cases?  Like, do they test with the code to the right of the cursor cut off? Or if the cursor is at teh end of the file?  etc.?